pennylane/fourier/reconstruct.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains a function that computes the fourier series equivalent to
+a quantum expectation value."""
+from functools import wraps
+from inspect import signature
+import warnings
+
+# import numpy as np
+import pennylane as qml
+from pennylane import numpy as np
+
+
+def _reconstruct_equ(fun, num_frequency, fun_at_zero=None):
+    r"""Reconstruct a univariate Fourier series with consecutive integer
+    frequencies, using trigonometric interpolation and equidistant shifts.
+
+    This technique is based on
+    `Dirichlet kernels <https://en.wikipedia.org/wiki/Dirichlet_kernel>`_, see
+    `Vidal and Theis (2018) <https://arxiv.org/abs/1812.06323>`_ or
+    `Wierichs et al. (2021) <https://arxiv.org/abs/2107.12390>`_.
+
+    Args:
+        fun (callable): Function to reconstruct
```

Comment:
I assume that we require `fun` to be periodic?

Reply:
also, might be good to specify that the reconstruction applies only to the first positional argument of `fun`, and that `fun` is `float -> float`.

Reply:
Ah, yeah all this information is kind of subsummed in "univariate Fourier series" above, but I'll also include it in the `Args` section :+1: